### PR TITLE
feat(nav drawer): hierachical navigation sample with igx-tree

### PR DESCRIPTION
**Pull request by kdinev**
Reusing the `Routes` configuration to populate a tree for hierarchical nav.

### CODE REVIEW

**Review by dobromirts — DISMISSED** [review]

File: src/app/index/index.component.ts [hunk 1]
```diff
@@ -132,6 +138,11 @@ export class IndexComponent implements OnInit, AfterViewInit {
                 (route: any) => route.path === event.url)[0];
             if (routeItem) {
                 this.selectedDisplayName = routeItem.displayName;
+                const chilren = this.tree.findNodes(this.selectedDisplayName);
```
Comment (zdrawku) [inline src/app/index/index.component.ts:141]:
```suggestion
                const children = this.tree.findNodes(this.selectedDisplayName);
```

File: src/app/index/index.component.ts [hunk 2]
```diff
@@ -132,6 +138,11 @@ export class IndexComponent implements OnInit, AfterViewInit {
...
+                this.tree.deselectAll();
+                if (chilren?.length) {
```
Comment (zdrawku) [inline src/app/index/index.component.ts:143]:
```suggestion
                if (children?.length) {
```

File: src/app/index/index.component.ts [hunk 3]
```diff
@@ -132,6 +138,11 @@ export class IndexComponent implements OnInit, AfterViewInit {
...
+                    chilren[0].selected = true;
```
Comment (zdrawku) [inline src/app/index/index.component.ts:144]:
```suggestion
                    children[0].selected = true;
```

File: src/app/index/index.component.ts [hunk 4]
```diff
@@ -155,19 +166,41 @@ export class IndexComponent implements OnInit, AfterViewInit {
...
                 (routeItem) => routeItem.displayName === loadedRouteItem.displayName)[0];
 
-            this.toggleParent('header' + loadedParentItem.name);
-            document.getElementById('child' + loadedChildItem.displayName).scrollIntoView();
+            const parents = this.tree.findNodes(loadedParentItem.name);
+            if (parents?.length) {
+                parents[0].expanded = true;
+                parents[0].nativeElement.scrollIntoView();
+            }
+            const chilren = this.tree.findNodes(loadedChildItem.displayName);
```
Comment (zdrawku) [inline src/app/index/index.component.ts:174]:
```suggestion
            const children = this.tree.findNodes(loadedChildItem.displayName);
```

File: src/app/index/index.component.ts [hunk 5]
```diff
@@ -155,19 +166,41 @@ export class IndexComponent implements OnInit, AfterViewInit {
...
+            if (chilren?.length) {
```
Comment (zdrawku) [inline src/app/index/index.component.ts:175]:
```suggestion
            if (children?.length) {
```

File: src/app/index/index.component.ts [hunk 6]
```diff
@@ -155,19 +166,41 @@ export class IndexComponent implements OnInit, AfterViewInit {
...
+                chilren[0].selected = true;
```
Comment (zdrawku) [inline src/app/index/index.component.ts:176]:
```suggestion
                children[0].selected = true;
```

**Review by zdrawku — COMMENTED** [review]
Other than that it looks good to me.

**Comment by ChronosSF** [comment]
@kdinev , this looks all kinds of broken on Stackblitz for some reason. One is that the generator config seems to lack RouterModule import in order for the stackblitz app to recognize the `routerLink` directive.  However, even with it, the tree doesn't expand and throws null injector errors.

**Comment by kdinev** [comment]
@zdrawku @dobromirts Could you guys please take a look at @ChronosSF's comment?

**Review by ChronosSF — APPROVED** [review]

**Pull request merged**